numpy/ma/extras.py

```diff
@@ -1173,9 +1173,9 @@ def _covhelper(x, y=None, rowvar=True, allow_masked=True):
                 # Define some common mask
                 common_mask = np.logical_or(xmask, ymask)
                 if common_mask is not nomask:
-                    x.unshare_mask()
-                    y.unshare_mask()
                     xmask = x._mask = y._mask = ymask = common_mask
+                    x._sharedmask = False
+                    y._sharedmask = False
```

Comment:
Just thought I would remove it, unshare mask does nothing except setting _sharedmask to false, since the mask is overwritten anyway. Removing this means that the `__setitem__` call to unshare_mask is the only remaining call in numpy. Coupld possibly be deprecated itself when this is done.